pkg/cloudprovider/provider/kubevirt/provider.go

```diff
@@ -377,14 +381,19 @@ func (p *provider) parseOSImageSource(primaryDisk kubevirttypes.PrimaryDisk, nam
 	switch imageSource(osImageSource) {
 	case httpSource:
 		return &cdiv1beta1.DataVolumeSource{HTTP: &cdiv1beta1.DataVolumeSourceHTTP{URL: osImage}}, nil
+	case dockerSource, ociArchiveSource:
+		return &cdiv1beta1.DataVolumeSource{Registry: &cdiv1beta1.DataVolumeSourceRegistry{URL: &osImage}}, nil
 	case pvcSource:
 		if namespaceAndName := strings.Split(osImage, "/"); len(namespaceAndName) >= 2 {
 			return &cdiv1beta1.DataVolumeSource{PVC: &cdiv1beta1.DataVolumeSourcePVC{Name: namespaceAndName[1], Namespace: namespaceAndName[0]}}, nil
 		}
 		return &cdiv1beta1.DataVolumeSource{PVC: &cdiv1beta1.DataVolumeSourcePVC{Name: osImage, Namespace: namespace}}, nil
 	default:
 		// handle old API for backward compatibility.
-		if _, err = url.ParseRequestURI(osImage); err == nil {
+		if srcURL, err := url.ParseRequestURI(osImage); err == nil {
```

Comment:
We don't have to add this for backward compatibility, the plan is to remove this eventually.

Reply:
afaict, the [dashboard even for v2.23](https://github.com/kubermatic/dashboard/blob/74fb62248747acd8727bea957c972319dff5b7ed/modules/api/pkg/resources/machine/common.go#L536-L537) passes just the osImage URI and the source is an empty string. Happy to remove this after it's addressed in the dashboard but at the moment, without this and with https://github.com/kubermatic/dashboard/releases/tag/v2.23.0-rc.0, machine-controller was still trying to interpret `docker://` URLs as HTTP source which resulted in broken `VM`.

Reply:
As discussed internally we are going to deliver this feature to KKP 2.24 that gives us time to prepare API and UI. 
So this can be removed from backward compatibility section.

Reply:
We had another discussion about it, seems like all coutnerarguments are weak against speedin up the local kkp installer. As finally agreed let's keep registry type also in backward compatibility. So I am fine with the change.